core/sail/memory/src/test/java/org/eclipse/rdf4j/sail/memory/MemoryRDFStarSupportTest.java

```diff
@@ -7,19 +7,25 @@
  *******************************************************************************/
 package org.eclipse.rdf4j.sail.memory;
 
+import java.io.File;
+
 import org.eclipse.rdf4j.repository.RDFStarSupportTest;
 import org.eclipse.rdf4j.repository.Repository;
 import org.eclipse.rdf4j.repository.sail.SailRepository;
+import org.junit.jupiter.api.io.TempDir;
 
 /**
  * @author jeen
  *
  */
 public class MemoryRDFStarSupportTest extends RDFStarSupportTest {
 
+	@TempDir
+	File tempDir;
+
 	@Override
 	protected Repository createRepository() {
-		return new SailRepository(new MemoryStore());
+		return new SailRepository(new MemoryStore(tempDir));
```

Comment:
Would it make sense to duplicate this so we have one class with persistence and another without?

Reply:
We can do that easily enough, but I'm not sure if it adds anything: the tests without persistence test exactly the same thing, only without persistence. If the test _with_ persistence succeeds, the one without will, also.